### ISSUE TITLE
Reduce the number of run test when pushing to a PR

### ISSUE THREAD

**Issue by delapuente** [Qiskit/qiskit-terra#1072]
### What is the expected enhancement?
Reduce the number of test run when pushing to a PR by focusing on a reference environment (Python 3.5 + Linux) and add a cron task to perform all the checks once per day.

**Comment by delapuente** [comment]
@diego-plan9 @mtreinish I know some of you have reservations about this and I think those are fair points. Would you mind to make them explicit here?

Just to be clear, my point is that, with the current configuration, we are running the same tests over and over, even thinking that the major part of our codebase is platform independent. I was hoping to catch platform-specific bugs in the daily task. But I know you both share the fear of those bugs being unattended and not fixed (for which I open [this other issue](https://github.com/Qiskit/qiskit-terra/issues/926)). Perhaps it is a better idea to merge parallel testing first and throw money to AppVeyor and Travis before removing testing scenarios.

If you feel strongly against these changes, I'm OK. Feel free to close the issues.

**Comment by diego-plan9** [comment]
I wouldn't call them reservations - rather, observations!! There are several topics around that might have been also touched in other issues or PRs, but for me it's clear that we have a non-ideal situation during day-to-day working in the repository and indeed we should place emphasis in finding ways of making it smoother.

But I think there is another important force: how confident we are that we are not introducing bugs at each stage, _and_ how we keep that confidence as high as possible in order to catch them effectively. I tend to think of it as some sort of "numbers game". If we think that "passing _all_ (including online and slow) the current tests, in all platforms and python versions", is a 100% confidence (which is not, of course) my concern is that we might end up reducing that % even more with subsequent changes. Which turns the problem into "setting up a worfklow for dealing with bugs caught not during a PR, but some time after it".

Actually, I'm positive that we can handle the setting of that workflow (even though `master` has been failing for three days without getting much attention lately), and that we can keep a reasonable % of confidence by being more smarts on which tests we run and where we run it - but an effort (more organizative than technical) needs to be done. On a side note, I think introducing the travis `cron` job would be good for some scenarios regardless of what is decided for the PR/master setup: for example, the running of `slow_test`s which is very much needed. 

So basically I'm personally fine with whatever @delapuente and @mtreinish agree upon, as long as ultimately we end up being able to catch bugs and correct them effectively :bug:

**Comment by ajavadia** [comment]
AppVeyor has an option called "rolling build" which ensures only the latest commit to a PR branch is built. I just turned it on for this project, which will hopefully improve those Windows build times.

**Comment by diego-plan9** [comment]
Nice!

**Comment by diego-plan9** [comment]
Closing as we agreed on keeping the stages numerous during PRs.
